### ISSUE TITLE
Universal install-type

### ISSUE THREAD

**Issue by eisterman** [gcad3d/gcad3d#13]
Hi, I'm an ArchLinux user and I wanted to use this software but while I was compiling the source I noticed how the possibility of compiling without creating a deb or a rpm is not considered.

Is it possible in the future to have a compile option without mandatory packaging?

**Comment by gcad3d** [comment]
The first step is to colpile all then an additional step creates a deb or rpm. ???
See README. Which commands did you use ?
Franz.

**Comment by alex-fu27** [comment]
I wanted to try `gcad3d` today, but I believe that I ran into the same problem:

```
[alex@alexnb APP]$ ./do debug_OFF
gcad_dir_dev=|/mnt/c/Users/Alexander/projects/sw/gcad3d/|
gcad_dir_bin=|/mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/|
gcad_dir_dev:  /mnt/c/Users/Alexander/projects/sw/gcad3d/
gcad_dir_bin:  /mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/
[alex@alexnb APP]$ ./do complete
gcad_dir_dev=|/mnt/c/Users/Alexander/projects/sw/gcad3d/|
gcad_dir_bin=|/mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/|
gcad_dir_dev:  /mnt/c/Users/Alexander/projects/sw/gcad3d/
gcad_dir_bin:  /mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/
gcad_dir_dev=|/mnt/c/Users/Alexander/projects/sw/gcad3d/|
gcad_dir_bin=|/mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/|
delete all gcad-objs (*.o)
gcad_dir_dev=|/mnt/c/Users/Alexander/projects/sw/gcad3d/|
gcad_dir_bin=|/mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/|
gcad_dir_dev:  /mnt/c/Users/Alexander/projects/sw/gcad3d/
gcad_dir_bin:  /mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/
which: no rpm in (/home/alex/bin:/home/alex/.local/bin:/home/alex/bin:/home/alex/.local/bin:/home/alex/bin:/home/alex/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/uti:/opt/cuda/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
instTyp = |deb|
../gui_version_set.sh: Zeile 19: dpkg: Kommando nicht gefunden.
**** gui_version_set.sh ERROR gtk2 not installed
gcad_dir_dev=|/mnt/c/Users/Alexander/projects/sw/gcad3d/|
gcad_dir_bin=|/mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/|
gcad_dir_dev:  /mnt/c/Users/Alexander/projects/sw/gcad3d/
gcad_dir_bin:  /mnt/c/Users/Alexander/projects/sw/gcad3d/build/gcad3d/Linux_x86_64/
which: no rpm in (/home/alex/bin:/home/alex/.local/bin:/home/alex/bin:/home/alex/.local/bin:/home/alex/bin:/home/alex/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/uti:/opt/cuda/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
instTyp = |deb|
../gui_version_set.sh: Zeile 37: dpkg: Kommando nicht gefunden.
**** gui_version_set.sh ERROR gtk3 not installed
****** ERROR build gui-gtk3 *******
```

The problem is that in fact very few Linux distributions use `dpkg` as a package manager, and not all use the Debian package format or RPM. This may work on your home PC and most Debian PCs, but to reach more users, you need a build configuration system that has a more distribution-independent way of checking for dependencies. Like `cmake`, for example.

Your current build scripts are in fact quite confusing to me. As far as I understand, I have to set the environment variable `DIR_DEV` to the *parent directory* of the location where I cloned the repository. So the first assumption you made is that everyone will clone the repository into a directory called `gcad3d`. So for example naming it `gdac3d-git` or `test_gcad3d` is impossible. This also raises fears in me that your script could change something in my `~/projects/sw/`.

To return to @eisterman's issue: I belive that he saw the output I got and thought the script needed `rpm` for building because of the output `which: no rpm in ...`

**Comment by gcad3d** [comment]
Hallo,

concerning:
>  very few Linux distributions use dpkg as a package manager ..
- what else would be better ?

>  fears in me that your script could change something in my ~/projects/sw/.
- why should there be a difference to test_gcad3d or any other directory ?
- why not eg ../projects/sw/test_/gcad3d/  ??

> To return to @eisterman's issue
- we do not have knowledge of eisterman's issue


Mit freundlichen Grüßen

Franz Reiter


---------------------------------------------------------
CAD-CAM Services Ing. Franz Reiter
Alfons Petzoldgasse 41
A 1230 Wien
Tel. 0043 (0)699 10202284
email franz.reiter@cadcam.co.at
http://www.cadcam.co.at
http://www.gcad3d.org
---------------------------------------------------------

**Comment by alex-fu27** [comment]
> we do not have knowledge of eisterman's issue

Yes! That is because he did not post any logs because he probably gave up on trying to build gcad3d quickly after discovering that you did not consider that somebody might want to build gcad3d on a different system than whatever you are using.

> what else would be better?

Please, please make people able to build gcad3d from source without having any package manager installed at all.

**Comment by gcad3d** [comment]
- if you do:

# build:
./do complete
# start (test)
./do
# copy into system
sudo -E ./do install_test_deb

# user-start
cd
gcad3d

- where is the problem ??

**Comment by alex-fu27** [comment]
The only thing I can do for you is to again post a copy of my shell output. This time I attached it to the comment.

See [test.log](https://github.com/gcad3d/gcad3d/files/3744037/test.log)

**Comment by gcad3d** [comment]
dpkg -l checks if the development-libs of gtk2 or gtk3 are installed.
We did not find a useful replacement,
so we must add dpkg to the software required for build.

**Comment by gcad3d** [comment]
build-scripts are updated; no usage of dpkg or rpm; see howto at -
https://gcad3d.org/doc/install_en.txt

short:
# start commandline (eg gnome-terminal)
# activate home-directory
cd
# create basedirectory for installation (arbitrary name - eg devel)
mkdir devel
cd devel
# get all files into ~/devel/gcad3d/*
git clone https://github.com/gcad3d/gcad3d.git
# goto dev-dir
cd ${HOME}/devel/gcad3d/src/APP
# build complete
./do complete
# start gcad3d
./do
